types/rollup/index.d.ts

```diff
@@ -55,26 +55,26 @@ export interface BundleOptions {
 	 * Whether to include the 'use strict' pragma at the top of generated non-ES6 bundles.
 	 * Strictly-speaking (geddit?), ES6 modules are always in strict mode, so you shouldn't disable this without good reason.
 	 */
-	useStrict?: boolean
+	strict?: boolean
 }
 
 export interface GenerateOptions extends BundleOptions {
 	/** Whether to generate a sourcemap. If true, the return value from `bundle.generate(...)` will include a map property */
-	sourceMap?: boolean
+	sourcemap?: boolean
 	/**
 	 * The location of the generated bundle. If this is an absolute path, all the sources paths in the sourcemap will be relative to it.
-	 * The map.file property is the basename of sourceMapFile, as the location of the sourcemap is assumed to be adjacent to the bundle.
+	 * The map.file property is the basename of sourcemapFile, as the location of the sourcemap is assumed to be adjacent to the bundle.
 	 */
-	sourceMapFile?: string
+	sourcemapFile?: string
 }
 
 export interface WriteOptions extends BundleOptions {
-	/** The file to write to. If `options.sourceMap === true`, two files will be created – `dest` and `dest + '.map`. */
-	dest: string
-	/** If `true`, a separate sourcemap file will be created. If `'inline'`, the sourcemap will be appended to the resulting dest file as a data URI. */
-	sourceMap?: boolean | 'inline'
-	/** This option is unnecessary, as it defaults to the value of dest. */
-	sourceMapFile?: string
+	/** The file to write to. If `options.sourcemap === true`, two files will be created – `file` and `file + '.map`. */
+	file: string
```

Comment:
this is wrong, it’s <code>options<em>.output</em>.file</code>:

https://github.com/rollup/rollup/blob/b949eb08169115ff66648838cbc4833379bf9440/src/utils/deprecateOptions.js#L50

Reply:
As far a I understand the part "[Changes to the API](https://gist.github.com/Rich-Harris/d472c50732dab03efeb37472b08a3f32#changes-to-the-api)" and the source code the deprecateOptions script is only used for the config file. Not for the API. 

I assumed the type definition was meant to be used for the API only.

Reply:
ah i see. so we have to separate the write options from the config file options.

Reply:
If anyone was curious the [rollup/index.js file](https://github.com/rollup/rollup/blob/master/src/rollup/index.js#L66) contains the deprecation checks for the API.